lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -69,17 +72,12 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
             "path": f"/{indices}/_search",
             "kwargs": {
                 "json": {
+                    "size": 10000,
                     "query": {
                         "bool": {
                             "filter": [
                                 {"match": {"run_data_parent": run_id}},
-                                {
-                                    "dis_max": {
-                                        "queries": {
-                                            "regexp": {"directory": f"{parent}[^/]+"}
-                                        }
-                                    }
-                                },
+                                {"regexp": {"directory": f"{parent}[^/]+"}},
                             ],
                             "must_not": {
                                 "regexp": {"directory": f"{parent}[^/]+/[^/]+"}
```

Comment:
Does the value of `parent` always contain a trailing slash?  If not, I think this query will match things that it shouldn't (like other directories with a suffix in their names), and it won't match things that it _should_ (like the target directory itself!); if it does, then I don't think this query will match the target directory -- it will only match subdirectories of it.  (The example in the docstring shows it without a trailing slash.)

I think we need some code somewhere upstream of here which removes (or adds...but I think "removes" is the right approach) the trailing slash (but which also handles the "root" request properly), so that this code can count on having a reasonable value to work with (and so that I as a reader know what to expect...).

Also, at line 53, we're using `get()` to fetch the value of `parent`, but we're not detecting the case where the result is `None`, and I don't think we want to proceed if it is....  (That is, I _think_ it should never be `None`, and so we should be accessing it as `json_data["parent"]` instead of using `get()`.)

Finally, I still think that the `"size"` value should be a symbolic constant.  🙂